src/Razor/src/Microsoft.VisualStudio.LanguageServer.ContainedLanguage/DefaultLSPRequestInvoker.cs

```diff
@@ -109,60 +104,46 @@ internal class DefaultLSPRequestInvoker : LSPRequestInvoker
         var response = await _languageServiceBroker.RequestAsync(
-            textBuffer,
-            capabilitiesFilter,
-            languageServerName,
-            method,
-            ParameterFactory,
+            new DocumentRequest<TIn, TOut>()
+            {
+
+                TextBuffer = textBuffer,
+                LanguageServerName = languageServerName,
+                ParameterFactory = _ => parameters,
+                Method = method,
+            },
             cancellationToken);
-#pragma warning restore CS0618 // Type or member is obsolete
 
         if (response is null)
         {
             return null;
         }
 
-        var responseBody = default(TOut);
-        if (response.Response is not null)
-        {
-            responseBody = response.Response.ToObject<TOut>(_serializer);
-        }
-
-        var reinvocationResponse = new ReinvocationResponse<TOut>(response.LanguageClientName, responseBody);
+        var reinvocationResponse = new ReinvocationResponse<TOut>(languageServerName, response);
         return reinvocationResponse;
     }
 
-    private async Task<IEnumerable<ReinvokeResponse<TOut>>> RequestMultipleServerCoreAsync<TIn, TOut>(string method, string contentType, Func<JToken, bool> capabilitiesFilter, TIn parameters, CancellationToken cancellationToken)
+    private async Task<IEnumerable<ReinvokeResponse<TOut>>> RequestMultipleServerCoreAsync<TIn, TOut>(string method, TIn parameters, CancellationToken cancellationToken)
         where TIn : notnull
     {
         if (string.IsNullOrEmpty(method))
         {
             throw new ArgumentException("message", nameof(method));
         }
 
-        var serializedParams = JToken.FromObject(parameters);
-
-#pragma warning disable CS0618 // Type or member is obsolete
-        var clientAndResultTokenPairs = await _languageServiceBroker.RequestMultipleAsync(
-            new[] { contentType },
-            capabilitiesFilter,
-            method,
-            serializedParams,
+        var requests = _languageServiceBroker.RequestAllAsync(
+            new GeneralRequest<TIn, TOut>() { LanguageServerName = null, Method = method, Request = parameters},
             cancellationToken).ConfigureAwait(false);
-#pragma warning restore CS0618 // Type or member is obsolete
 
-        // a little ugly - tuple deconstruction in lambda arguments doesn't work - https://github.com/dotnet/csharplang/issues/258
-        var results = clientAndResultTokenPairs.Select((clientAndResultToken) => clientAndResultToken.Item2 is not null ? new ReinvokeResponse<TOut>(clientAndResultToken.Item1, clientAndResultToken.Item2.ToObject<TOut>(_serializer)!) : default);
+        List<ReinvokeResponse<TOut>> responses = new();
+        await foreach (var response in requests)
+        {
+            // No callers actually use the languageClient when handling the response.
+            responses.Add(new ReinvokeResponse<TOut>(default!, response.response!));
```

Comment:
What is the first parameter? If its the language server client name (ie, a string), please use `null` and provide the parameter name.

Also, should we doing a null check on `response.response`?

Reply:
Fixed.  response.response is that actual return, which in practice should never be null (and also, removing the method soon enough)